src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -449,13 +456,22 @@ async def fetch_with_coursier(request: CoursierFetchRequest) -> FallibleClasspat
     # TODO: Loading this per JvmArtifact.
     lockfile = await Get(CoursierResolvedLockfile, CoursierResolveKey, request.resolve)
 
```

Comment:
NOTE: The following code is for validating **user** lockfiles, and is necessarily done per `JvmArtifact`, because of how we bubble a resolved classpath up to the consuming code.

Reply:
That'd be good to put in a comment. To clarify, that is because this is the rule for `CoursierFetchRequest`?